### PR TITLE
Eliminate duplicate saving of DAgger demonstrations

### LINKED ISSUE #402: dagger demos saved twice
Hi,
I noticed a larger than expected number of demonstrations per round and I think i found the problem:

The demonstrations are saved both here:
https://github.com/HumanCompatibleAI/imitation/blob/9ca1e441ac4ee2c43f96a57b044cf625cab8b31a/src/imitation/algorithms/dagger.py#L635
and here:
https://github.com/HumanCompatibleAI/imitation/blob/9ca1e441ac4ee2c43f96a57b044cf625cab8b31a/src/imitation/algorithms/dagger.py#L259

Is this the expected behaviour?

### PR DESCRIPTION
**Pull request by AdamGleave**
Fixes #402. One-line change in DAgger to remove line duplicated from `InteractiveTrajectoryCollector`. Some more involved changes to the tests to let us check number of demo episodes saved is correct. In particular, needed to switch to `Pendulum-v1` from `CartPole-v0` to benefit from fixed length episodes letting us back out expected number of episodes from total timesteps.

### CODE REVIEW

File: tests/algorithms/test_dagger.py
Comment (qxcv) [inline tests/algorithms/test_dagger.py:115]:
```suggestion
    # roll out 5 * venv.num_envs episodes (Pendulum-v1 has 200 timestep episodes)
```

(make env version consistent with test declaration)

**Review by qxcv — COMMENTED** [review]
This for fixing this! One question and one suggestion.

File: tests/algorithms/test_dagger.py
Comment (AdamGleave) [reply tests/algorithms/test_dagger.py:59]: We currently only actually have an expert policy serialized for CartPole. I think this decision was originally to just minimize the number of binary blobs we commit to Git.

The tests I introduced re: # of files saved don't actually need an expert policy, any policy will do, but I agree this is confusing and error-prone.

#405 will change this to generate expert policies on-demand and save them in CircleCI cache, at which point we can just get rid of this code. I've added a TODO to that effect. We can either wait until that is merged, or merge this PR and clean it up in #405

Comment (qxcv) [reply tests/algorithms/test_dagger.py:61]: Sure. Given there's a comment, I'm fine with merging this now and then fixing it later in #405.

**Comment by AdamGleave** [comment]
Now the changes in https://github.com/HumanCompatibleAI/imitation/pull/405 have been merged, I've made more significant changes to `test_dagger.py` to switch everything to using the fixed-horizon Pendulum environment. This is easier than having a mixture of CartPole and Pendulum. Only reason I didn't do this before was we didn't have Pendulum expert policy, but now we do. Requesting re-review on these changes (which only affect test code).

File: tests/algorithms/test_dagger.py
Comment (qxcv) [inline tests/algorithms/test_dagger.py:274]: May help to add some context:

```suggestion
        # want to adjust this check if changing seed. Pendulum return can range
        # from -1,200 to -130 (approx.), per Figure 3 in this PDF (on page 3):
        # https://arxiv.org/pdf/2106.09556.pdf
```

**Review by qxcv — APPROVED** [review]
LGTM, feel free to accept or reject suggested comment (I was using it to verify that ranges were reasonable).

**Pull request merged**
